src/search.js

```diff
@@ -85,16 +85,32 @@ export default class Search {
     if (Array.isArray(this.query.filter) && this.query.filter.length === 0) {
       return results;
     }
-    const docIds = Object.keys(this.searchFields(this.query.filter));
-    const filtered = {};
 
-    Object.keys(results).forEach((id) => {
-      if (docIds.indexOf(id) !== -1) {
-        filtered[id] = results[id];
-      }
+    const resultIds = Object.keys(results);
+
+    if (resultIds.length === 0) {
+      return results;
+    }
+
+    const queriesArray = Array.isArray(this.query.filter) ? this.query.filter : [this.query.filter];
```

Comment:
Quibble, but should this be `filtersArray`?

Reply:
I was thinking of them as filter _queries_, but I can see how that might be harder to track!